### PR TITLE
Add basic support for datatypes

### PR DESCRIPTION
**Pull request by loichuder**
h5wasm returns `null` when encountering a datatype while reading a file so that consumers cannot really figure out what is the actual entity.

I tried to make things clearer by having a specific class `Datatype`. So far, this class is dead simple and only contains `type = 'Datatype'` but we could imagine enhancing it in the future to include to actual datatype value if needed.

### CODE REVIEW

File: test/make_test_files.py [hunk 1]
```diff
@@ -12,11 +12,13 @@
 f.create_dataset("string_arr", (2,), dtype=sarray_type)
 f["string_arr"][:] = np.array([b"hello", b"there"] * 2 * 2).reshape((2,2,2))
 
-f.create_dataset("compound", (2,), dtype=[('floaty', array_type), ('stringy',sarray_type)])
-f['compound'][:] = np.array([(f['float_arr'][0], f['string_arr'][0]), (f['float_arr'][1], f['string_arr'][1])], dtype=f['compound'].dtype)
+f.create_dataset("compound", (2,), dtype=[("floaty", array_type), ("stringy",sarray_type)])
+f["compound"][:] = np.array([(f["float_arr"][0], f["string_arr"][0]), (f["float_arr"][1], f["string_arr"][1])], dtype=f["compound"].dtype)
 
-f.create_dataset("bool", data=[False, True, True, False], shape=(2,2))
+f.create_dataset("bool", data=[[False, True], [True, False]], shape=(2,2))
```
Comment (loichuder) [inline test/make_test_files.py:18]: Made the `data` match the `shape`

Comment (bmaranville) [reply test/make_test_files.py:18]: If it passes the tests still, sounds good!

**Review by bmaranville — APPROVED** [review]
Thanks for the very complete PR - including tests!

File: test/make_test_files.py (hunk 1 above)
Comment (loichuder) [reply test/make_test_files.py:18]: Yeah, in the end, I don't think it was necessary :sweat_smile: 

Well, at least, the passing tests show that `.value` and `.to_array` are robust enough

**Pull request merged**